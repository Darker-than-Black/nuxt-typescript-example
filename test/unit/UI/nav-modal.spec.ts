import VueTestInstance from '~/test/factory/vue-instance'
import NavModal from '~/components/UI/nav-modal.vue'

describe('nav-modal.vue', () => {
  const slotContent: string = '<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam</p>'
  const wrapper = VueTestInstance(NavModal, {
    options: {
      slots: { default: slotContent }
    }
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check slot of component', () => {
    expect(wrapper.find('.nav-modal p').html()).toEqual(slotContent)
  })
})
