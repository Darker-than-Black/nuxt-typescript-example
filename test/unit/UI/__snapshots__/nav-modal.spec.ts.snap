// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav-modal.vue should match the snapshot 1`] = `
<div class="nav-modal">
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam</p>
</div>
`;
